lib/kadai-core/src/main/java/io/kadai/spi/task/internal/TaskDistributionManager.java

```diff
@@ -0,0 +1,77 @@
+package io.kadai.spi.task.internal;
+
+import io.kadai.common.api.KadaiEngine;
+import io.kadai.common.api.exceptions.InvalidArgumentException;
+import io.kadai.common.internal.util.LogSanitizer;
+import io.kadai.common.internal.util.SpiLoader;
+import io.kadai.spi.task.api.TaskDistributionProvider;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class TaskDistributionManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TaskDistributionManager.class);
+  private final List<TaskDistributionProvider> taskDistributionProviderList;
+
+  public TaskDistributionManager(KadaiEngine kadaiEngine) {
+    taskDistributionProviderList = SpiLoader.load(TaskDistributionProvider.class);
+    for (TaskDistributionProvider taskDistributionProvider : taskDistributionProviderList) {
+      taskDistributionProvider.initialize(kadaiEngine);
+      LOGGER.info(
+          "Registered TaskDistribution provider: {}",
+          taskDistributionProvider.getClass().getName());
+    }
+
+    if (taskDistributionProviderList.isEmpty()) {
+      LOGGER.info(
+          "No Custom TaskDistribution Provider found. Running with "
+              + "DefaultTaskDistributionProvider");
+    }
+  }
+
+  public TaskDistributionProvider getProviderByName(String name) {
+    return taskDistributionProviderList.stream()
+        .filter(provider -> provider.getClass().getSimpleName().equals(name))
+        .findFirst()
+        .orElseThrow(
+            () ->
+                new InvalidArgumentException(
+                    String.format("The distribution strategy '%s' does not exist.", name)));
+  }
+
+  public Map<String, List<String>> distributeTasks(
+      List<String> taskIds,
+      List<String> destinationWorkbasketIds,
+      Map<String, Object> additionalInformation,
+      String distributionStrategyName) {
+
+    Map<String, List<String>> newTaskDistribution;
+
+    if (distributionStrategyName != null) {
+
+      TaskDistributionProvider taskDistributionProvider =
+          this.getProviderByName(distributionStrategyName);
+
+      String sanitizedDistributionStrategyName =
+          LogSanitizer.stripLineBreakingChars(distributionStrategyName);
+      LOGGER.info("Using TaskDistributionProvider: {}", sanitizedDistributionStrategyName);
+
+      newTaskDistribution =
+          taskDistributionProvider.distributeTasks(
+              taskIds, destinationWorkbasketIds, additionalInformation);
+    } else {
+      LOGGER.info("No distribution strategy specified. Using default distribution.");
+      newTaskDistribution =
+          new DefaultTaskDistributionProvider()
```

Comment:
why don't you add the default provider to the list of provider? In that case you don't even need a if-then-else block?